modules/http/http-worker.c

```diff
@@ -203,27 +201,33 @@ _format_request_headers(HTTPDestinationWorker *self, LogMessage *msg)
        * backward compatibility when batching was introduced, however I think
        * this should eventually be removed */
 
-      headers = _add_header(headers,
-                            "X-Syslog-Host",
-                            log_msg_get_value(msg, LM_V_HOST, NULL));
-      headers = _add_header(headers,
-                            "X-Syslog-Program",
-                            log_msg_get_value(msg, LM_V_PROGRAM, NULL));
-      headers = _add_header(headers,
-                            "X-Syslog-Facility",
-                            syslog_name_lookup_name_by_value(msg->pri & LOG_FACMASK, sl_facilities));
-      headers = _add_header(headers,
-                            "X-Syslog-Level",
-                            syslog_name_lookup_name_by_value(msg->pri & LOG_PRIMASK, sl_levels));
+      _add_header(self->request_headers,
+                  "X-Syslog-Host",
+                  log_msg_get_value(msg, LM_V_HOST, NULL));
+      _add_header(self->request_headers,
+                  "X-Syslog-Program",
+                  log_msg_get_value(msg, LM_V_PROGRAM, NULL));
+      _add_header(self->request_headers,
+                  "X-Syslog-Facility",
+                  syslog_name_lookup_name_by_value(msg->pri & LOG_FACMASK, sl_facilities));
+      _add_header(self->request_headers,
+                  "X-Syslog-Level",
+                  syslog_name_lookup_name_by_value(msg->pri & LOG_PRIMASK, sl_levels));
     }
 
   for (l = owner->headers; l; l = l->next)
-    headers = curl_slist_append(headers, l->data);
+    list_append(self->request_headers, l->data);
 
   if (owner->auth_header)
-    headers = _append_auth_header(headers, owner);
+    _append_auth_header(self->request_headers, owner);
+
+  HttpHeaderRequestSignalData signal_data =
+  {
+    .request_headers = self->request_headers,
+    .request_body = self->request_body
+  };
 
-  return headers;
+  EMIT(owner->super.super.super.super.signal_slot_connector, signal_http_header_request, &signal_data);
```

Comment:
It would be nice to encapsulate this line together with the creation of signal data into a function: `collect_rest_headers(...)`, or similar.

Reply:
done